test/e2e/gateway_api_test.go

```diff
@@ -363,12 +367,21 @@ func testGatewayAPIResourcesProtection(t *testing.T) {
 		t.Run(tc.name, func(t *testing.T) {
 			// Verify that GatewayAPI CRD creation is forbidden.
 			for i := range testCRDs {
-				if err := tc.kclient.Create(context.Background(), testCRDs[i]); err != nil {
-					if !strings.Contains(err.Error(), tc.expectedErrMsg) {
-						t.Errorf("unexpected error received while creating CRD %q: %v", testCRDs[i].Name, err)
+				if err := wait.PollUntilContextTimeout(context.Background(), 2*time.Second, 30*time.Second, false, func(ctx context.Context) (bool, error) {
+					if err := tc.kclient.Create(ctx, testCRDs[i]); err != nil {
+						if kerrors.IsAlreadyExists(err) {
+							// VAP was disabled and re-enabled at the beginning of the test.
+							t.Log("VAP may not be processed yet; retrying...")
```

Comment:
Better to log what we observe rather than logging what we guess the reason to be.  As Joe Friday would say, all we want are the facts.  
```suggestion
							t.Logf("Failed to create CRD %q: %v; retrying...", testCRDs[i].Name, err)
```

Reply:
Done.